torch/csrc/sparse/CusparseLtKernels.cpp

```diff
@@ -0,0 +1,316 @@
+#include <ATen/cuda/CUDAContext.h>
+#include <ATen/cuda/CUDADataType.h>
+#include <c10/core/ScalarType.h>
+#include <c10/util/Half.h>
+#include <cusparse.h>
+#include <torch/csrc/sparse/cusparseLt.h>
+#include <torch/custom_class.h>
+#include <torch/torch.h>
+
+#define CHECK_CUDA(func)                                    \
+  {                                                         \
+    cudaError_t status = (func);                            \
+    TORCH_CHECK(                                            \
+        status == cudaSuccess,                              \
+        "CUDA API failed at line %d with error: %s (%d)\n", \
+        __LINE__,                                           \
+        cudaGetErrorString(status),                         \
+        status)                                             \
+  }
+
+#define CHECK_CUSPARSE(func)                                    \
+  {                                                             \
+    cusparseStatus_t status = (func);                           \
+    TORCH_CHECK(                                                \
+        (status == CUSPARSE_STATUS_SUCCESS),                    \
+        "CUSPARSE API failed at line %d with error: %s (%d)\n", \
+        __LINE__,                                               \
+        cusparseGetErrorString(status),                         \
+        status);                                                \
+  }
+namespace torch {
+namespace ao {
+namespace pruning {
+
+// create a container that holds relevant data for cusparselt matmul
+struct CusparseLt : public torch::CustomClassHolder {
```

Comment:
Do you have info about CustomClassHolder? Who maintains that code?

Reply:
I don't know too much about CustomClassHolder, I think James Reed wrote this [tutorial](https://pytorch.org/tutorials/advanced/torch_script_custom_classes.html#conclusion)

Reply:
Given that this is a torchscript specific feature and that it is unmaintained. I would recommend to use pybind directly here to get a solution that is maintainable.

Reply:
Gotcha, do you know if there's a good PR I can use as reference?

Reply:
I don't have a PR but you can see code examples here for example: https://github.com/pytorch/pytorch/blob/f0360c99ca396c77e11f38d6d7d449c660d59b6e/torch/csrc/autograd/init.cpp#L140